src/gradualizer_lib.erl

```diff
@@ -83,58 +84,86 @@ reverse_graph(G) ->
     from_edges(maps:keys(G), [ {J, I} || {I, Js} <- maps:to_list(G), J <- Js ]).
 
 
+get_user_type_definition(Types, Anno, Name, Args) ->
```

Comment:
To get fewer and easier-to-remember parameters, why not keep the `user_type` tuple here? as in

```suggestion
get_user_type_definition({user_type, Anno, Name, Args}, Types) ->
```

A short comment about the function and the parameters is always nice to have. Types is a map from name to type, right? (Or a spec.)